helix-term/src/lib.rs

```diff
@@ -22,3 +25,26 @@ fn true_color() -> bool {
 fn true_color() -> bool {
     true
 }
+
+/// Function used for filtering dir entries in the various file pickers.
+///
+/// We always want to ignore the .git directory, otherwise if
+/// `ignore` is turned off, we end up with a lot of noise
+/// in our picker.
+/// We also ignore symlinks that point inside the current directory
+/// if `dedup_links` is enabled.
+fn filter_entry(entry: &DirEntry, root: &Path, dedup_links: bool) -> bool {
+    if entry.file_name() != ".git" {
+        return false;
+    }
+
+    if dedup_links && entry.path_is_symlink() {
+        return entry
+            .path()
+            .canonicalize()
+            .ok()
+            .map_or(false, |path| !path.starts_with(&root));
+    }
+
+    true
+}
```

Comment:
I believe the logic in here around filtering the `.git` folder is backwards. https://docs.rs/ignore/0.4.20/ignore/struct.WalkBuilder.html#method.filter_entry Might want to put the comments closer to their usage in the code. Might give the function a more expressive name.
```suggestion
/// Function used for filtering dir entries in the various file pickers.
fn ignore_git_and_simlinks(entry: &DirEntry, root: &Path, dedup_links: bool) -> bool {
    // We always want to ignore the .git directory, otherwise if
    // `ignore` is turned off, we end up with a lot of noise
    // in our picker.
    if entry.file_name() == ".git" {
        return false;
    }

    // We also ignore symlinks that point inside the current directory
    // if `dedup_links` is enabled.
    if dedup_links && entry.path_is_symlink() {
        return entry
            .path()
            .canonicalize()
            .ok()
            .map_or(false, |path| !path.starts_with(&root));
    }

    true
}

```

Reply:
Thanks for pointing out the `.git` mistake to me. I refactored this after I was done and messed up the `&&` to `if` conversion. I should probably stop writing code at 3am. This function is used both in `commands.rs` and `ui/mod.rs` so this is as close to the usage as it gets.